lib/gc/Transforms/GPU/Pipeline.cpp

```diff
@@ -63,22 +63,24 @@ void populateGPUPipeline(OpPassManager &pm,
   pm.addPass(createBufferizationToMemRefPass());
 
   pm.addNestedPass<func::FuncOp>(createForallToParallelLoopPass());
+  pm.addNestedPass<func::FuncOp>(createGpuMapParallelLoopsPass());
+  pm.addNestedPass<func::FuncOp>(createParallelLoopToGpuPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addNestedPass<func::FuncOp>(createAllocsToSLM());
   pm.addNestedPass<func::FuncOp>(createLinalgToXeGPU(
       {/*kTile=*/16, /*stages=*/1, /*dpasTiles=*/{8, 16, 16}}));
+  pm.addPass(createCSEPass());
 
   pm.addNestedPass<func::FuncOp>(createConvertLinalgToLoopsPass());
   pm.addPass(xegpu::createXeGPUFoldAliasOps());
   pm.addPass(memref::createFoldMemRefAliasOpsPass());
-  pm.addNestedPass<func::FuncOp>(createGpuMapParallelLoopsPass());
-  pm.addNestedPass<func::FuncOp>(createParallelLoopToGpuPass());
 
   imex::InsertGPUAllocsOptions insertGPUAllocsOption{
       /*clientAPI*/ "opencl", /*inRegions*/ false,
       /*isUsmArgs*/ pipelineOpts.isUsmArgs};
   pm.addNestedPass<func::FuncOp>(
       imex::createInsertGPUAllocsPass(insertGPUAllocsOption));
   pm.addPass(createGpuKernelOutliningPass());
-  pm.addPass(createCanonicalizerPass());
```

Comment:
Canonicalizer converts `vector.from_elements [%val, %val, ... %val]` into `vector.splat %val` that causes the `imex::ConvertGPUXToSPIRVPass` to fail (it seems it doesn't support `vector.splat`). So removed canonicalizer